op-batcher/batcher/driver.go

```diff
@@ -945,3 +919,136 @@ func logFields(xs ...any) (fs []any) {
 	}
 	return fs
 }
+
+type ChannelStatuser interface {
+	isFullySubmitted() bool
+	isTimedOut() bool
+	LatestL2() eth.BlockID
+	MaxInclusionBlock() uint64
+}
+
+type SyncActions struct {
+	clearState      *eth.BlockID
+	blocksToPrune   int
+	channelsToPrune int
+	waitForNodeSync bool
+	blocksToLoad    [2]uint64 // the range [start,end] that should be loaded into the local state.
+	// NOTE this range is inclusive on both ends, which is a change to previous behaviour.
+}
+
+func (s SyncActions) String() string {
+	return fmt.Sprintf(
+		"SyncActions{blocksToPrune: %d, channelsToPrune: %d, waitForNodeSync: %t, clearState: %v, blocksToLoad: [%d, %d]}", s.blocksToPrune, s.channelsToPrune, s.waitForNodeSync, s.clearState, s.blocksToLoad[0], s.blocksToLoad[1])
+}
+
+// computeSyncActions determines the actions that should be taken based on the inputs provided. The inputs are the current
+// state of the batcher (blocks and channels), the new sync status, and the previous current L1 block. The actions are returned
+// in a struct specifying the number of blocks to prune, the number of channels to prune, whether to wait for node sync, the block
+// range to load into the local state, and whether to clear the state entirely.
+func computeSyncActions[T ChannelStatuser](newSyncStatus eth.SyncStatus, prevCurrentL1 eth.L1BlockRef, blocks queue.Queue[*types.Block], channels []T, l log.Logger) SyncActions {
```

Comment:
I'm not sure it makes a lot of sense here, since we don't have "steps" and multiple pipeline stages? Open to renaming in general though if we don't like `computeSyncActions`.